command-expansion-server/src/app.ts

```diff
@@ -35,46 +36,87 @@ app.post("/put-dictionary", async (req, res) => {
   console.log(`command-lib generated - path: ${commandDictionaryPath}`);
 
   const sqlExpression = `
-      insert into command_dictionary (command_types, mission, version)
-      values ($1, $2, $3)
-      on conflict (mission, version) do update
-      set command_types = $1
-      returning id;
-    `;
+    insert into command_dictionary (command_types, mission, version)
+    values ($1, $2, $3)
+    on conflict (mission, version) do update
+    set command_types = $1
+    returning id;
+  `;
 
   const { rows } = await db.query(sqlExpression, [
     commandDictionaryPath,
     parsedDictionary.header.mission_name,
     parsedDictionary.header.version,
   ]);
 
-  if (rows.length < 0) {
+  if (rows.length < 1) {
     console.error(`POST /dictionary: No command dictionary was updated in the database`);
-    res.status(400).json({ message: `POST /dictionary: No command dictionary was updated in the database` });
-  } else {
-    const id = rows[0].id;
-    res.status(200).json({ id });
+    res.status(500).send(`POST /dictionary: No command dictionary was updated in the database`);
+    return;
   }
+  const id = rows[0].id;
+  res.status(200).json({ id });
   return;
 });
 
-app.put("/expansion/:activityTypeName", async (req, res) => {
-  res.status(501).send("PUT /expansion: Not implemented");
+app.post('/get-expansion', async (req, res) => {
+  const expansionId = req.body.input.expansionId as string;
+  console.log(`POST /get-expansion: Retrieved expansion from database: id=${expansionId}`);
+
+  const { rows } = await db.query(`
+    SELECT expansion_logic
+    FROM expansion_rules
+    WHERE id = $1;
+  `, [
+    expansionId,
+  ]);
+
+  if (!rows.length) {
+    throw new ErrorWithStatusCode(`POST /get-expansion: No expansion with id=${expansionId}`, 404);
+  }
+
+  const expansionLogicTypescriptBase64 = Buffer.from(rows[0].expansion_logic).toString('base64');
+  res.status(200).json({
+    typescript: expansionLogicTypescriptBase64,
+  });
+  return;
+});
+
```

Comment:
Should we ignore the rate limiting CodeQL failure? I'm not sure how to accomplish rate limiting with PG or if we want to?